### PR TITLE
micronaut-security-basicauth fixes

### PR DESCRIPTION
**Pull request by burtbeckwith**
mostly the same as https://github.com/micronaut-projects/micronaut-guides/pull/456

### CODE REVIEW

File: guides/micronaut-security-basicauth/groovy/src/main/groovy/example/micronaut/DefaultAuthorizationExceptionHandlerReplacement.groovy [hunk 1]
```diff
@@ -3,22 +3,27 @@ package example.micronaut
...
 @Replaces(DefaultAuthorizationExceptionHandler) // <2>
 class DefaultAuthorizationExceptionHandlerReplacement extends DefaultAuthorizationExceptionHandler {
 
     @Override
-    protected MutableHttpResponse<?> httpResponseWithStatus(HttpRequest request, AuthorizationException exception) {
-        if (exception.isForbidden()) {
-            return HttpResponse.status(HttpStatus.FORBIDDEN)
+    protected MutableHttpResponse<?> httpResponseWithStatus(HttpRequest request,
+                                                            AuthorizationException e) {
+        if (e.isForbidden()) {
```
Comment (sdelamo) [inline guides/micronaut-security-basicauth/groovy/src/main/groovy/example/micronaut/DefaultAuthorizationExceptionHandlerReplacement.groovy:22]: I like the original flow without an else bracket better. I think it is nicer to only remove the return statement when it is the last sentence.

File: guides/micronaut-security-basicauth/java/src/main/java/example/micronaut/DefaultAuthorizationExceptionHandlerReplacement.java [hunk 2]
```diff
@@ -3,23 +3,26 @@
...
 @Replaces(DefaultAuthorizationExceptionHandler.class) // <2>
 public class DefaultAuthorizationExceptionHandlerReplacement extends DefaultAuthorizationExceptionHandler {
 
     @Override
-    protected MutableHttpResponse<?> httpResponseWithStatus(HttpRequest request, AuthorizationException exception) {
-        if (exception.isForbidden()) {
-            return HttpResponse.status(HttpStatus.FORBIDDEN);
+    protected MutableHttpResponse<?> httpResponseWithStatus(HttpRequest request,
+                                                            AuthorizationException e) {
+        if (e.isForbidden()) {
```
Comment (sdelamo) [inline guides/micronaut-security-basicauth/java/src/main/java/example/micronaut/DefaultAuthorizationExceptionHandlerReplacement.java:22]: I like the original flow without an else bracket better. I think it is nicer to only remove the return statement when it is the last sentence.

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: guides/micronaut-security-basicauth/java/src/main/java/example/micronaut/DefaultAuthorizationExceptionHandlerReplacement.java (hunk 2 above)
Comment (burtbeckwith) [reply guides/micronaut-security-basicauth/java/src/main/java/example/micronaut/DefaultAuthorizationExceptionHandlerReplacement.java:22]: did you mean this comment for the Kotlin version? I didn't remove a return statement here

File: guides/micronaut-security-basicauth/groovy/src/main/groovy/example/micronaut/DefaultAuthorizationExceptionHandlerReplacement.groovy (hunk 1 above)
Comment (burtbeckwith) [reply guides/micronaut-security-basicauth/groovy/src/main/groovy/example/micronaut/DefaultAuthorizationExceptionHandlerReplacement.groovy:22]: updated

File: guides/micronaut-security-basicauth/java/src/main/java/example/micronaut/DefaultAuthorizationExceptionHandlerReplacement.java (hunk 2 above)
Comment (sdelamo) [reply guides/micronaut-security-basicauth/java/src/main/java/example/micronaut/DefaultAuthorizationExceptionHandlerReplacement.java:22]: Sorry, it seems I miss commented here.

**Review by sdelamo — APPROVED** [review]

**Pull request merged**